lib/active_merchant/billing/gateways/pay_junction_v2.rb

```diff
@@ -110,6 +112,27 @@ def add_payment_method(post, payment_method)
         end
       end
 
+      def add_address(post, options)
+        if (address = options[:billing_address])
+          post[:billing] = {}
+          post[:billing][:firstName] = address[:first_name] if address[:first_name]
+          post[:billing][:lastName] = address[:last_name] if address[:last_name]
+          post[:billing][:companyName] = address[:company] if address[:company]
+          post[:billing][:phone] = address[:phone_number] if address[:phone_number]
+
+          post[:billing][:address] = {}
+          post[:billing][:address][:address] = address[:address1] if address[:address1]
+          post[:billing][:address][:city] = address[:city] if address[:city]
+          post[:billing][:address][:state] = address[:state] if address[:state]
+          post[:billing][:address][:country] = address[:country] if address[:country]
+          post[:billing][:address][:zip] = address[:zip] if address[:zip]
+        end
+      end
+
+      def get_state(address)
```

Comment:
Hmm, is this method called anywhere?  I'm _guessing_ it should be called up on line 126.

Reply:
For the `X-PJ-Application-Key`, that is the PayJunction specific access key that is added to the headers. You'll find it in Spreedly's Onepass and you should just add it to your fixtures assigned to the `api_key`.

Reply:
Good catch. I'm not sure I actually want that method. Was following Leila's Adyen example and I think that default to 'NA' is an Adyen gateway thing.

Reply:
Yes, confirming that Adyen is a bit finicky about requiring non-nil fields (e.g., "NA") to be sent for certain address information even if that information isn't available. Most gateways don't have those restrictions.